### ISSUE TITLE
Implement GROUP BY

### ISSUE THREAD

**Issue by stem-admin** [join-monster/join-monster#85]
Implement the GROUP BY strategy brainstormed earlier to subquery using the `sqlJoin` as the `where` and taking user-specified grouping columns and aggregate functions on the field.

**Comment by peterviergutz** [comment]
Hi,

I am developing a prototype for my current project and man, join-monster is a decent piece of software that can solve 80% of my problems out of the box. GroupBy is in the other 20%, so I was wondering why this feature was closed? 

I hacked a group by into the QueryRoot resolver by just collecting the fields to be grouped by in the context and appending groupby in the dbCall function of the demo - it works pretty well when you sqlBatch the join. So I was wondering if there's any blocker further down the road that I've missed? If not, a hint into the right direction how this ought to be tackled would be highly appreciated. I'd be happy to contribute to this great project :)

Thanks!

**Comment by acarl005** [comment]
Interesting.... do you have a sample or a fork I can check out? I ended up closing it because I couldn't get a consistent, efficient implementation that worked in all situations, e.g. after a `sqlJoin` or a `sqlBatch`. I am certainly open to the idea.

**Comment by peterviergutz** [comment]
>  I couldn't get a consistent, efficient implementation that worked in all situations

I agree, adding robust support without limitations will be hard, if possible at all. Please do not consider this a proper approach - it's just the result of a quick PoC if join-monster can solve problems in my current project - I added a groupBy hack to the demo [here](https://github.com/peterviergutz/join-monster-demo/tree/feature/groupBy), just 

```
npm install
npm start
```

Sample query: 

```
{
  users {
    id
    stats(from: "2016-01-01", to: "2017-08-02") {
      comments
      likes
    }
    posts {
      body
      stats(from: "2016-01-01", to: "2017-08-02") {
        comments
        likes
      }
    }
  }
}
```

General functionality is to populate `context.groupBy` in the `where()` of the `StatsConnector` by climbing up the nodes in the `sqlASTNode` and then appending the group by SQL in the `resolver()` of the QueryRoot.
